plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMergeSink.java

```diff
@@ -209,31 +210,27 @@ private FileWriter createParquetFileWriter(String path, List<DeltaLakeColumnHand
         try {
             Closeable rollbackAction = () -> fileSystem.deleteFile(path);
 
-            List<Type> parquetTypes = dataColumns.stream()
-                    .map(column -> {
-                        Type type = column.getType();
+            List<TypedColumn> columns = dataColumns.stream()
+                    .map(dataColumn -> {
+                        Type type = dataColumn.getType();
                         if (type instanceof TimestampWithTimeZoneType timestamp) {
                             verify(timestamp.getPrecision() == 3, "Unsupported type: %s", type);
-                            return TIMESTAMP_MILLIS;
+                            type = TIMESTAMP_MILLIS;
                         }
```

Comment:
Use a helper method to retrieve the type - e.g `getType(Type)` which can be then inlined in the `TypedColumn` constructor.

Reply:
Ok, extracted to `getSupportedType` static private method, since this type supporting relates to delta lake specific operation (but TypedColumn belongs to Parquet converter).